scripts/SHI/scpSHI_Jetz_complete.R

```diff
@@ -0,0 +1,396 @@
+#forest base map
+cube_GFW_TC <-
+  load_cube(stac_path = "https://io.biodiversite-quebec.ca/stac/",
+            limit = 1000,
+            collections = c("gfw-treecover2000"),
+            bbox = sf_ext_crs,
+            srs.cube = srs_cube,
+            spatial.res = spat_res,
+            temporal.res = "P1Y",
+            t0 = "2000-01-01",
+            t1 = "2000-12-31",
+            resampling = "bilinear")
+
+cube_GFW_TC_range <- cube_GFW_TC |>
+  gdalcubes::filter_pixel(paste0("data >=", forest_threshold)) |>#filter to forest threshold 0-100
+  gdalcubes::apply_pixel("data/data") # turn into a forest presence map
+
+r_GFW_TC_range <- cube_to_raster(cube_GFW_TC_range , format="terra") # convert to raster format
+
+r_suitability_map_rescaled <- terra::resample(r_suitability_map,r_GFW_TC_range,method="mode") #Adjust scale of suitability map
+r_GFW_TC_range_mask <- r_GFW_TC_range |>
+  terra::classify(rcl=cbind(NA,0)) |> # turn NA to 0
+  terra::mask(r_suitability_map_rescaled) # crop to suitability map
+
+print("========== Base forest layer downloaded ==========")
+
+# Download forest loss maps and create different layers for each year to remove from forest
+s_obj <- stac("https://io.biodiversite-quebec.ca/stac/")
+
+it_obj <- s_obj |>
+  stac_search(collections = "gfw-lossyear",
+              bbox = sf_ext) |>
+  get_request()
+
+st <- gdalcubes::stac_image_collection(it_obj$features, asset_names = c("data")) # create image collection
+
+v <- cube_view(srs = srs_cube, extent = list(t0 = "2000-01-01", t1 = "2000-12-31", # create cube according to area of interest
+                                             left = sf_ext_crs['xmin'], right = sf_ext_crs['xmax'],
+                                             top = sf_ext_crs['ymax'], bottom =  sf_ext_crs['ymin']),
+               dx=spat_res, dy=spat_res, dt="P1Y",
+               resampling = "mode", aggregation = "mode") # TO CHANGE to proportions
+
+times <- as.numeric(substr(v_time_steps[v_time_steps>2000],start=3,stop=4)) #get year of change by selected time step to mask map by year of change
+l_r_year_loss <- map(times, function(x) {
+  layer <- raster_cube(image_collection=st, view=v,
+                       mask=image_mask("data", values = times[times<=x],invert=T)) # to remove it to the following years
+
+  layer <- layer |>
+    stars::st_as_stars() |>
+    terra::rast()
+  return(layer)
+})
+
+#add background and mask to suitable area
+s_year_loss <- terra::classify(terra::rast(l_r_year_loss), rcl=rbind(c(NA,NA,0),c(1,Inf,1)),include.lowest=T)
+names(s_year_loss) <- paste0("Loss_",v_time_steps[v_time_steps>2000])
+s_year_loss_mask <- terra::mask(s_year_loss,r_GFW_TC_range_mask, maskvalues=1, inverse=TRUE)
+
+#update reference forest layer if t_0 different of 2000
+if(t_0!=2000){
+  r_GFW_TC_range_mask <- terra::classify(r_GFW_TC_range_mask - terra::subset(s_year_loss_mask,paste0("Loss_",t_0)),rcl=cbind(-1,0))
+}
+
```

Comment:
There should be a title here.  Not clear why habitat gain only for one year (start year?)

Reply:
I have this one:
#update reference forest layer if t_0 different of 2000

This step updates the reference layer (start year), because we only have data from 2000, but if the starting year is later than that we would need to calculate it

Reply:
Maybe add a comment in the code?